Sources/Atomic.swift

```diff
@@ -55,10 +48,12 @@ internal struct UnsafeAtomicState<State: RawRepresentable> where State.RawValue
 	///   - next: The state to transition to.
 	///
 	/// - returns: `true` if the transition succeeds. `false` otherwise.
-	internal func tryTransition(from expected: State, to next: State) -> Bool {
-		return OSAtomicCompareAndSwap32Barrier(expected.rawValue,
-		                                       next.rawValue,
-		                                       value)
+	internal mutating func tryTransition(from expected: State, to next: State) -> Bool {
+		return withUnsafeMutablePointer(to: &value) { valuePointer in
+			return OSAtomicCompareAndSwap32Barrier(expected.rawValue,
+			                                       next.rawValue,
+			                                       valuePointer)
+		}
```

Comment:
I did check the compiler source and generated assembly. `withUnsafeMutablePointer` uses `Builtin.addressof`, which would emit the in-place address if the variable has direct storage. IOW it is not atomic only if `self` is not a final stored property.